ovn_cluster.sh

```diff
@@ -195,75 +203,104 @@ function add-ovs-container-ports() {
+                (( ip_index += 1))
+
+                ip2=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name}-2 --ipaddress=${ip2}/${cidr}
+                echo "$name $ip2" >> _ovn_central_2
+                (( ip_index += 1))
+
+                ip3=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name}-3 --ipaddress=${ip3}/${cidr}
+                echo "$name $ip3" >> _ovn_central_3
+                (( ip_index += 1))
+
+                echo "$name ${REMOTE_PROT}:$ip1:6642,${REMOTE_PROT}:$ip2:6642,${REMOTE_PROT}:$ip3:6642" >> _ovn_remote
+            else
+                ip=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name} --ipaddress=${ip}/${cidr}
+                echo "$name ${REMOTE_PROT}:$ip:6642" >> _ovn_remote
+                (( ip_index += 1))
+            fi
+        done
 
         for name in "${GW_NAMES[@]}"; do
-            (( ip_index += 1))
             ip=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
             ./ovs-runc add-port $br $eth ${name} --ipaddress=${ip}/${cidr}
+            (( ip_index += 1))
         done
 
         if [ "$RELAY_COUNT" -gt 0 ]; then
+            echo -n > _ovn_relay_remotes
+
             relay_remotes=""
+            last_az="az1"
+
             for name in "${RELAY_NAMES[@]}"; do
-                (( ip_index += 1))
+                relay_az=$(echo $name | awk -F- '{print $3}')
+                if [ "$relay_az" != "$last_az" ]; then
+                    ovn_relay_remotes=$(echo $relay_remotes | cut -c 2-)
+                    echo "ovn-central-$last_az $ovn_relay_remotes" >> _ovn_relay_remotes
+                    relay_remotes=""
+                    last_az="$relay_az"
+                fi
+
                 ip=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
                 ./ovs-runc add-port $br $eth ${name} --ipaddress=${ip}/${cidr}
                 relay_remotes=$relay_remotes",${REMOTE_PROT}:$ip:6642"
+                (( ip_index += 1))
             done
-            orig_remotes=$(cat _ovn_remote)
-            echo "${relay_remotes}" | cut -c 2- > _ovn_remote
-            echo "${orig_remotes}" > _ovn_remote_main_db
+            ovn_relay_remotes=$(echo $relay_remotes | cut -c 2-)
+            echo "ovn-central-$last_az $ovn_relay_remotes" >> _ovn_relay_remotes
+
+            cat _ovn_remote > _ovn_remote_main_db
+            cat _ovn_relay_remotes > _ovn_remote
         fi
 
     else
         if [ "$OVN_DB_CLUSTER" = "yes" ]; then
-            (( ip_index += 2))
+            (( ip_index += 3*CENTRAL_COUNT))
+        else
+            (( ip_index += $CENTRAL_COUNT))
         fi
 
         if [ "$GW_COUNT" -gt 0 ]; then
             (( ip_index += $GW_COUNT))
         fi
 
         if [ "$RELAY_COUNT" -gt 0 ]; then
-            (( ip_index += $RELAY_COUNT))
+            (( ip_index += CENTRAL_COUNT*RELAY_COUNT))
         fi
```

Comment:
Partially related to this PR but because we're touching this part now, I think we can just remove this whole "else" block all together.  If we're not adding central components (`"$ovn_central" == "no"`) we might as well let the user try to figure out the right IPs (through the IP_START env var).

Reply:
@dceara looking again at this chunk of code I think it is necessary for _ovn-heater_ since we run `add-chassis` in _bringup-cluster.yml_ with `ovn_central=no`

Reply:
I suggested to remove this because ovn-heater can pass a proper IP_START value (that includes the already configured nodes (central, relay, chassis, etc).